### PR TITLE
Migrate to Jest

### PR DESCRIPTION
**Pull request by li-kai**
Moved tests to Jest

### CODE REVIEW

File: v3/__tests__/actions/moduleBank.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// @flow
+import type { FSA } from 'types/redux';
+
+import * as actions from 'actions/moduleBank';
+
+// TODO: Change to generated snapshot when migrated to Jest.
```
Comment (yangshun) [inline v3/__tests__/actions/moduleBank.js]: Can remove this TODO

File: v3/__tests__/actions/moduleBank.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+test('fetchModuleList should return a request action', () => {
+  const resultOfAction: FSA = actions.fetchModuleList();
+  expect(resultOfAction.type).toBe(actions.FETCH_MODULE_LIST);
```
Comment (yangshun) [inline v3/__tests__/actions/moduleBank.js]: If we are matching the whole action to its snapshot, there's no need to assert the action type.

File: v3/__tests__/actions/settings.js [hunk 3]
```diff
@@ -2,45 +2,44 @@
...
-test('should dispatch a select of a newStudent value', (t) => {
+test('should dispatch a select of a newStudent value', () => {
   const newStudent = false;
   const expectedResult: FSA = {
     type: actions.SELECT_NEW_STUDENT,
     payload: newStudent,
   };
   const resultOfAction: FSA = actions.selectNewStudent(newStudent);
-  t.deepEqual(resultOfAction, expectedResult);
+  expect(resultOfAction).toEqual(expectedResult);
```
Comment (yangshun) [inline v3/__tests__/actions/settings.js]: Why not use snapshots for all these?

File: v3/__tests__/actions/timetable.js [hunk 4]
```diff
@@ -1,15 +1,12 @@
...
 import type { FSA } from 'types/redux';
 import type { ModuleCode, Semester, Lesson } from 'types/modules';
-// import nock from 'nock';
-
-import test from 'ava';
 import * as actions from 'actions/timetables';
 import lessons from '../mocks/lessons-array.json';
 
 // see: https://github.com/reactjs/redux/blob/master/docs/recipes/WritingTests.md#example-1
 // TODO: write addModule test with nock and mockStore.
```
Comment (yangshun) [inline v3/__tests__/actions/timetable.js]: Is this TODO still relevant?

File: v3/package.json [hunk 5]
```diff
@@ -97,11 +96,20 @@
     "redux-logger": "^2.6.1",
     "redux-thunk": "^2.2.0"
   },
-  "ava": {
-    "babel": "inherit",
-    "require": [
-      "babel-register"
-    ]
+  "jest": {
```
Comment (yangshun) [inline v3/package.json]: Let's use `.jestconfig` and keep config out of the `package.json`

File: v3/__tests__/actions/timetable.js (hunk 4 above)
Comment (li-kai) [inline v3/__tests__/actions/timetable.js]: Yep apparently lol.

**Pull request merged**